src/Service/SchedulingService.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace App\Service;
+
+use App\Entity\Car;
+use App\Entity\Appointment;
+use App\Exception\DuplicateAppointmentException;
+use App\Repository\AppointmentRepository;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+class SchedulingService
+{
+    public function __construct(
+        private readonly AppointmentRepository $appointmentRepository,
+        private readonly EntityManagerInterface $entityManager,
+        )
+    {}
+
+
+    public function scheduleAppointment(Appointment $appointment): void
+{
+    $appointment->setScheduledAt($this->roundToNearestHalfHour($appointment->getScheduledAt()));
+
+    $existingAppointment = $this->appointmentRepository->findOneBy([
+        'car' => $appointment->getCar(),
```

Comment:
search only by scheduledAt, because it might happen someone else scheduled at that moment, so that time is not good for the shop